buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -677,13 +688,17 @@ void free() {
         }
 
         public AdaptiveByteBuf newBuffer() {
-            AdaptiveByteBuf buf = bufferQueue.poll();
-            if (buf == null) {
-                buf = new AdaptiveByteBuf(handle);
+            AdaptiveByteBuf buf;
+            if (handle == null) {
+                buf = EVENT_LOOP_LOCAL_BUFFER_POOL.get();
             } else {
-                buf.resetRefCnt();
-                buf.discardMarks();
+                buf = bufferQueue.poll();
+                if (buf == null) {
+                    buf = new AdaptiveByteBuf(handle);
+                }
             }
+            buf.resetRefCnt();
```

Comment:
I guess its not a big deal but we could just not do this when we allocate a new buffer...

Reply:
more of the times we end-up reusing some (hopefully) - but I'm ok to move it in the right branch too, LMK

Reply:
Anyway is not using a volatile operation, but a lazySet AFAIK, which is way cheaper